api/graphite.go

```diff
@@ -557,3 +460,95 @@ func (s *Server) metricsDeleteRemote(orgId int, query string, peer cluster.Node)
 
 	return resp.DeletedDefs, nil
 }
+
+// executePlan looks up the needed data, retrieves it, and then invokes the processing
+// note if you do something like sum(foo.*) and all of those metrics happen to be on another node,
+// we will collect all the indidividual series from the peer, and then sum here. that could be optimized
+func (s *Server) executePlan(orgId int, plan expr.Plan) ([]models.Series, error) {
+
+	type locatedDef struct {
+		def  idx.Archive
+		node cluster.Node
+	}
+	minFrom := uint32(math.MaxUint32)
+	var maxTo uint32
+	//locatedDefs[request][<def.id>]locatedDef
+	locatedDefs := make(map[expr.Req]map[string]locatedDef)
+
+	// note that different patterns to query can have different from / to, so they require different index lookups
+	// e.g. target=movingAvg(foo.*, "1h")&target=foo.*
+	// note that in this case we fetch foo.* twice. can be optimized later
+	for _, r := range plan.Reqs {
+		series, err := s.findSeries(orgId, []string{r.Query}, int64(r.From))
```

Comment:
afaict each call to `findSeries()` will wait until every backend peer has responded before it returns. so it might make sense to also parallelize the calls to `findSeries()` in a similar way like it parallelizes the backend calls, or do you think that might blow up into too many threads?